components/cartItem/CartItem.vue

```diff
@@ -0,0 +1,155 @@
+<script setup lang="ts">
+import type { Schemas } from '@shopware/api-client/api-types';
+import {toast} from '../ui/toast';
+import {ApiClientError} from '@shopware/api-client';
+import {getProductUrl} from '@shopware-pwa/helpers-next';
+
+const props = withDefaults(
+    defineProps<{
+      cartItem?: Schemas['LineItem'];
```

Comment:
Just a quick question. Is this field really optional or is it always required? If second, we should remove the `?` and just make it a required prop. This makes it also more visible when working iwht components

```suggestion
      cartItem: Schemas['LineItem'];
```